### DISCUSSION TITLE
Set message ID when emitting to Kafka topic.

### DISCUSSION THREAD

**Discussion by chainhead** [quarkusio/quarkus#29332] [Q&A]
I want to work with this [project](https://quarkus.io/guides/kafka-reactive-getting-started). I want to set message ID when writing the REST payload to Kafka topic. How do I set message ID when doing `quoteRequestEmitter.send(uuid.toString())`?

[a comment by an automated account is not shown]

**Reply by chainhead** [reply]
Perhaps, explore [this](https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/3.4/kafka/kafka.html#_sending_keyvalue_pairs)?

**Comment by ozangunalp** [comment]
You can use `KafkaRecord.of(key, value)` to build a Kafka record for outgoing messages.

**Reply by chainhead** [reply]
Is `KafkaRecord.of` an extension of `Record.of` ?

**Reply by ozangunalp** [reply]
No `KafkaRecord.of` creates a `Message` (OutgoingKafkaRecord) whereas `Record` is a regular payload. In both cases Kafka Connector will set key and value for sending the `ProducerRecord`.
